libraries/abstractions/common_io/test/test_iot_adc.c

```diff
@@ -82,6 +89,17 @@ TEST_SETUP( TEST_IOT_ADC )
  */
 TEST_TEAR_DOWN( TEST_IOT_ADC )
 {
+    /* Release the ADC hardware during teardown */
+    int32_t lRetVal;
+
+    lRetVal = iot_adc_close( xAdcHandle );
+
+    if (lRetVal == IOT_ADC_NOT_OPEN)
+    {
+        return;
```

Comment:
I prefer removing this return, and just guarding the assert with an if statement, as this is against the coding standard.

Reply:
I would say we can simply just close without checking the return status:

```c
/* Intentionally not check the return status. */
iot_adc_close( xAdcHandle );
```

This is an example why I like using "Test_Protect()" to solving such cascading failures. Then the open/close logic can be contained in each test case clearly.

Reply:
I think I will do a quick re-spin of the test cases to go with the "test_protect()" solution that you prefer so that the look and feel remain the same.

Reply:
Really appreciate that.